ism330dhcx driver: Fixed most mcuastyle errors

cparata opened the pull request:
Fixed most mcuastyle errors in the source and header files.

Signed-off-by: Carlo Parata <carlo.parata@st.com>

avisconti commented on ism330dhcx_reg.h:
```diff
@@ -2836,12 +2836,12 @@ typedef union
+int32_t ism330dhcx_write_reg(stmdev_ctx_t *ctx, uint8_t reg,
+                             uint8_t *data,
+                             uint16_t len);
 
```
Ciao Carlo,

Can you possibly put this change in a separate commit? I'd prefer to have this change separated from the rest of warnings. Thanks! (rest is absolutely fine!)

cparata replied:
Ok, now it should be fine

cparata committed:
ism330dhcx_reg.h: Fixed compilation warning with MDK-ARM

The __weak attribute must be removed from the function prototypes to
avoid a compilation warning with MDK-ARM toolchain.

Signed-off-by: Carlo Parata <carlo.parata@st.com>

cparata committed:
ism330dhcx driver: Fixed most mcuastyle errors

Fixed most mcuastyle errors in the source and header files.

Signed-off-by: Carlo Parata <carlo.parata@st.com>

avisconti reviewed (approved):
Perfect, thanks!
